src/package-request.js

```diff
@@ -136,6 +138,30 @@ export default class PackageRequest {
   }
 
   /**
+   * If pattern:
+   * - contains : it will return because is something like '<protocol>:<something else' or an url
+   * - is a file it will prepend file:
+   * otherwise it will return
+   */
+  async normalizeRange(pattern: string): Promise<string> {
+    if (pattern.includes(':')) {
+      return Promise.resolve(pattern);
+    }
+
+    if (await fs.exists(path.join(this.config.cwd, pattern))) {
```

Comment:
After some over-thinking we can do something like this:

```
    if (pattern.includes(':') ||
      pattern.includes('@') ||
      PackageRequest.getExoticResolver(pattern)) {
      return Promise.resolve(pattern);
    }
```
this should give us enough confidence that the pattern may be a file if we go ahead in the method. Do you like it? Do you think is enough?